kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/RollingUpdateProcessServiceIntegrationTest.java

```diff
@@ -204,54 +218,127 @@ public void testStartProcessInDifferentDeploymentWithContainerId() throws Except
 
         tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
-        assertEquals(1, tasks.size());
+        assertThat(tasks).hasSize(1);
         task = tasks.get(0);
-        assertEquals(CONTAINER_ID_101, task.getContainerId());
+        assertThat(task.getContainerId()).isEqualTo(CONTAINER_ID_101);
 
         taskClient.completeAutoProgress(CONTAINER_ALIAS, task.getId(), USER_YODA, null);
 
-        // let's start another instance of the old version of container
+        // let's start ano  ther instance of the old version of container
         Long processInstanceIdV3 = processClient.startProcess(CONTAINER_ID, PROCESS_ID_EVALUATION, parameters);
 
-        assertNotNull(processInstanceIdV3);
-        assertTrue(processInstanceIdV3.longValue() > 0);
+        assertThat(processInstanceIdV3).isNotNull();
+        assertThat(processInstanceIdV3).isGreaterThan(0);
 
         processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, processInstanceIdV3);
-        assertNotNull(processInstance);
-        assertEquals(CONTAINER_ID, processInstance.getContainerId());
+        assertThat(processInstance).isNotNull();
+        assertThat(processInstance.getContainerId()).isEqualTo(CONTAINER_ID);
 
         tasks = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
-        assertEquals(2, tasks.size());
+        assertThat(tasks).hasSize(2);
         task = tasks.get(0);
-        assertEquals(CONTAINER_ID, task.getContainerId());
+        assertThat(task.getContainerId()).isEqualTo(CONTAINER_ID);
 
         taskClient.completeAutoProgress(CONTAINER_ALIAS, task.getId(), USER_YODA, null);
 
     }
-}
+
+    @Test
+    public void testDifferentGAVsWithAlias() throws Exception {
+        Long oldPid = processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION);
+        assertThat(oldPid).isNotNull();
+        assertThat(oldPid).isGreaterThan(0);
+
+        ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ALIAS, oldPid);
+        assertThat(processInstance).isNotNull();
+        assertThat(processInstance.getContainerId()).isEqualTo(CONTAINER_ID);
+
+        // Create a container with completely different GAV
+        ReleaseId differentReleaseId101 = new ReleaseId("org.kie.server.different", "different-gav-definition-project",
+                                                        "1.0.1.Final");
+        KieContainerResource containerResource = new KieContainerResource(CONTAINER_ID_101, differentReleaseId101);
+        containerResource.setContainerAlias(CONTAINER_ALIAS);
+        client.createContainer(CONTAINER_ID_101, containerResource);
+
+        if(configuration.isRest()) {
+            assertClientException(
+                    () -> processClient.startProcess(CONTAINER_ALIAS, PROCESS_ID_EVALUATION_2), 404, MessageFormat.format(PROCESS_DEFINITION_NOT_FOUND, PROCESS_ID_EVALUATION_2, CONTAINER_ALIAS));
+
```

Comment:
@MarianMacik yes that was my intention so that I only bring into this pr the minimal amount of your work so that you can update and add all the other good tests you did. hope thats ok.